src/components/Player/Player.jsx

```diff
@@ -55,14 +29,28 @@ class RequestLayer extends React.Component {
   }
 
   componentWillUpdate(nextProps) {
-    if (this.props.params.accountId !== nextProps.params.accountId) {
+    if (this.props.params.accountId !== nextProps.params.accountId || this.props.location.key !== nextProps.location.key) {
       getData(nextProps);
     }
   }
 
   render() {
-    return <Player {...this.props} />;
+    const accountId = this.props.params.accountId;
+    const loading = this.props.loading;
+    const location = this.props.location;
+    const routeParams = this.props.routeParams;
+    const info = this.props.routeParams.info || 'overview';
+    const page = playerPages(accountId).find(page => page.name.toLowerCase() === info);
+    return (
+      <div>
+        <div className={styles.header}>
+          <PlayerHeader playerId={accountId} location={location} />
+          <TabBar info={info} tabs={playerPages(accountId)} />
+        </div>
+        {!loading && page ? page.content(accountId, routeParams, location) : <Spinner />}
```

Comment:
This won't work since load state is determined per page rather than at player level.  Need to determine spinner state at page level.